doc/Language/classtut.pod6

```diff
@@ -694,6 +623,41 @@ making dinner
 eating dinner. NOM!
 =end code
 
+
+X<|type object>
+X<|DEFINITE>
+X<|.DEFINITE>
+=head2 A word on types
+
+Declaring a class creates a new I<type object> which, by default, is installed
+into the current package (just like a variable declared with C<our> scope).
+This type object is an "empty instance" of the class. For example, types such as
+C<Int> and C<Str> refer to the type object of one of the Raku built-in
+classes. One can call methods on these type objects. So there is nothing special
+with calling the C<new> method on a type object.
+
+You can use the C<.DEFINITE> method to find out if what you have is an instance
+or a type object:
+
+=begin code
+say Int.DEFINITE; # OUTPUT: «False␤» (type object)
+say 426.DEFINITE; # OUTPUT: «True␤»  (instance)
+
+class Foo {};
+say Foo.DEFINITE;     # OUTPUT: «False␤» (type object)
+say Foo.new.DEFINITE; # OUTPUT: «True␤»  (instance)
+=end code
+
+In function signatures one can use so called type "smileys" to only accept instances
+or type objects:
+
+=begin code
+multi foo (Int:U) { "It's a type object!" }
+multi foo (Int:D) { "It's an instance!"   }
+say foo Int; # OUTPUT: «It's a type object!␤»
+say foo 42;  # OUTPUT: «It's an instance!␤»
+=end code
+
```

Comment:
The types section was moved here from right below the class keyword introduction and given a heading in https://github.com/Raku/doc/pull/4047/commits/d78116f76e5ed801616eeadaee5be8366a55dbf6.